client/handler.go

```diff
@@ -221,6 +297,64 @@ func (h *Handler) updateClient(ctx context.Context, c *Client) error {
 	return nil
 }
 
+// swagger:route PUT /dyn-clients/{id} public updateDynOAuth2Client
+//
+// Update an OAuth 2.0 Client
+//
+// Update an existing OAuth 2.0 Client. If you pass `client_secret` the secret will be updated and returned via the API. This is the only time you will be able to retrieve the client secret, so write it down and keep it safe.
+//
+// OAuth 2.0 clients are used to perform OAuth 2.0 and OpenID Connect flows. Usually, OAuth 2.0 clients are generated for applications which want to consume your OAuth 2.0 or OpenID Connect capabilities. To manage ORY Hydra, you will need an OAuth 2.0 Client as well. Make sure that this endpoint is well protected and only callable by first-party components.
+//
+//     Consumes:
+//     - application/json
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       200: oAuth2Client
+//       500: genericError
+func (h *Handler) UpdateDynamicRegistration(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	var c Client
+
+	if err := json.NewDecoder(r.Body).Decode(&c); err != nil {
+		h.r.Writer().WriteError(w, r, errorsx.WithStack(err))
+		return
+	}
+
+	var secret string
+	if len(c.Secret) > 0 {
+		secret = c.Secret
+	}
+	if _, err := h.r.ClientManager().Authenticate(r.Context(), ps.ByName("id"), []byte(secret)); err != nil {
```

Comment:
This will not allow us to update the client secret because `authenticate` will always fail. I guess it would be more common to require Basic Authorization or an access token with `sub === c.ID` here instead.